packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,58 @@
+import { PluginOption } from "vite";
+import upath from "upath";
+import { fileURLToPath } from "url";
+const currentDir = upath.dirname(fileURLToPath(import.meta.url));
+const packagesDir = upath.resolve(currentDir, "../..");
+
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    async resolveId(id, importer) {
+      if (!importer) {
+        return;
+      }
+      if (isImportedByStudio(importer)) {
+        return;
+      }
+      if (isStyleFile(id)) {
+        const resolvedId = (
+          await this.resolve(id, importer, { skipSelf: true })
+        )?.id;
+        if (resolvedId) {
+          return addQueryParameter(resolvedId, "inline");
+        }
+        throw new Error(
+          "Styling file could not be resolved by yext-studio-style-plugin"
+        );
+      }
+    },
+  };
+}
+
+function isImportedByStudio(importer: string) {
+  const unixImporter = upath.toUnix(importer);
+  return !!(
+    unixImporter.match(/.*\/node_modules\/@yext\/studio.*/) ||
+    importer.startsWith(packagesDir)
+  );
+}
+
+function isStyleFile(id: string) {
+  const idWithoutQueryParams = id.split("?")[0];
+  return !!idWithoutQueryParams.endsWith(".css");
+}
+
+function addQueryParameter(id: string, queryParameter: string) {
+  if (id.match(new RegExp(`.*\\?(.*&)?${queryParameter}`))) {
```

Comment:
what are the `\\` doing in this regex? I would have expected only one backslash to escape the question mark?

Reply:
Yeah, I was confused about that too at first. Since I'm writing the regex as a string, the backslash itself needs to be escaped.  That escaped backslash escapes the question mark.